clang-tools-extra/clangd/CodeComplete.cpp

```diff
@@ -452,6 +455,30 @@ struct CodeCompletionBuilder {
                           });
   }
 
+  // With all-scopes-completion, we can complete enum constants of scoped
+  // enums, in which case the completion might not be visible to Sema.
+  // So, if there's a using declaration for the enum class, manually
+  // drop the qualifiers.
+  void stripNamespaceForEnumConstantIfUsingDecl(const Symbol &IndexResult,
+                                                DeclContext *SemaDeclCtx) {
+    if (IndexResult.SymInfo.Kind != index::SymbolKind::EnumConstant)
+      return;
+    for (auto *Ctx = SemaDeclCtx; Ctx; Ctx = Ctx->getParent())
```

Comment:
can we pre-calculate the set of using decls point at enums once, and just check if enum we have at hand matches any of those? as performing this search for every completion item, especially considering how big can some enums can get, might result in quite a performance penalty.